crate/server/src/database/database_traits.rs

```diff
@@ -1,49 +1,27 @@
 use async_trait::async_trait;
-use cloudproof_findex::{
-    db_interfaces::{redis::FindexTable, rest::UpsertData},
-    reexport::cosmian_findex::{
-        TokenToEncryptedValueMap, TokenWithEncryptedValueList, Tokens, ENTRY_LENGTH, LINK_LENGTH,
-    },
-};
+
+use cosmian_findex::{Address, MemoryADT, ADDRESS_LENGTH};
 use cosmian_findex_structs::{EncryptedEntries, Permission, Permissions, Uuids};
 use uuid::Uuid;
 
 use crate::error::result::FResult;
 
-#[async_trait]
-pub(crate) trait FindexTrait: Sync + Send {
+use super::redis::WORD_LENGTH;
+
+pub(crate) trait FindexMemoryTrait:
```

Comment:
INFO : to use dyn capabalities we should assemble all required traits into one super trait